client/python/client.py

```diff
@@ -0,0 +1,284 @@
+import base64
+import time
+from urllib import response
+import grpc
+import uuid
+import threading
+from enum import Enum, auto
+from typing import Optional
+
+from armada.client import event_pb2
+from armada.client import event_pb2_grpc
+from armada.client import queue_pb2
+from armada.client import queue_pb2_grpc
+from armada.client import usage_pb2
+from armada.client import usage_pb2_grpc
+from armada.client import submit_pb2
+from armada.client import submit_pb2_grpc
+from k8s.io.api.core.v1 import generated_pb2 as core_v1
+from k8s.io.apimachinery.pkg.api.resource import generated_pb2 as api_resource
+
+class AuthMethod(Enum):
+        Anonymous = auto()
+        Basic = auto()
+        OpenId = auto()
+        Kerberos = auto()
+
+class AuthData:
+    def __init__(self, method: AuthMethod = AuthMethod.Anonymous,
+        username: Optional[str] = None, password: Optional[str] = None, # for BasicAuth
+        oidc_token: Optional[str] = None,
+        # TODO add options for kerberos authentication
+    ):
+
+        self.method = method
+
+        if self.method == AuthMethod.Anonymous:
+            pass
+        elif self.method == AuthMethod.Basic:
+            if not (username and password):
+                raise Exception("need username and password for Basic auth")
+            self.username = username
+            self.password = password
+        elif self.method == AuthMethod.OpenId:
+            if not oidc_token:
+                raise Exception("need oidc_token for OpenId auth")
+        elif self.method == AuthMethod.Kerberos:
+            pass
+
+class GrpcBasicAuth(grpc.AuthMetadataPlugin):
```

Comment:
I think it would be nice to explain this a bit more.  We went into the depths of GRPC docs to understand this so we should add a comment and maybe a link to where we discovered this.

Reply:
Done in c45a48d86baef3d1271e85d5d1d1709402c191d8